### PR TITLE
Started implementation of temperature, top_k and logits for local models

### PR DESCRIPTION
**Pull request by karemr2024**
- Removed hardcoded temperature, topk and logits
- Modified GeminiExecutor implementation to initialize temperature, topk and logits
- Modified atomic.rs config struct to include temperature, topk and logits
- Added the default values in case GeminiExecutor is initialized without new arguments
- Added test for new version of GeminiExecutor

### CODE REVIEW

**Comment by andthattoo** [comment]
Thanks for the contribution!

We can't merge this unless the newly added

```rust
    pub temperature: Option<f64>,   // Add temperature field
    pub top_k: Option<i32>,         // Add top_k field
    pub logits: Option<bool>, 
```

fields to Config struct are usable by each provider including OpenAI, Gemini, OpenRouter and Ollama.

Also logits are not supported by Ollama yet (see [issue](https://github.com/ollama/ollama/issues/2415))

Only option is to work vLLM out for compute nodes so logits are supported.  Maybe we can handle temp and top_k for all providers and keep them in the struct but remove logits until we have vLLM support.
